### ISSUE TITLE
clangd `UnusedIncludes: Strict` false positive when only concepts used from headers

### ISSUE THREAD

**Issue by chriselrod** [llvm/llvm-project#60702]
If I have a header `concepts.hpp` defining a concept
```cpp
#pragma once
#include <concepts>
#include <cstddef>

template <typename T>
concept HasSize = requires(T t) {
                    { t.size() } -> std::convertible_to<std::size_t>;
                  };
```
and a file `main.cpp` using the concept
```cpp
#include "concepts.hpp" // clangd unused include warning
#include <vector>

static_assert(HasSize<std::vector<int>>);
```
`clangd` will issue a warning in `main.cpp` that the header is unused, and suggest the automated action of removing that header.

Doing so of course results in errors.

**Comment by kadircet** [comment]
Thanks! Looks like this is because `TypeConstraint`s and `ConceptSpecializationExpr`s are not handled in our ast traversal. we should make sure we're visiting those and marking the underlying concept decl as used. a minimal example that covers both cases:
```
template <typename T>
concept C = true;

template <C T> struct Foo{};
static_assert(C<int>);
```

**Comment by danakj** [comment]
I noticed this today when I attempted to use clangd on a C++20 codebase as well.

**Comment by kadircet** [comment]
should be fixed with https://github.com/llvm/llvm-project/commit/ac2d2652db8de9ea2b750dd2bf1232941039dffe
